drivers/adc/ltc2358/ltc2358.c

```diff
@@ -0,0 +1,170 @@
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <stdlib.h>
+#include "ltc2358.h"
+#include "errno.h"
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Create 24-bit configuration word for the 8 channels.
+ * @param channel - Channel number.
+ * @param config_number - Configuration number for the channel.
+ * @param config_word - 24-bit config word created.
+ */
+void ltc2358_create_config_word(uint8_t channel, uint8_t config_number,
+				uint32_t *config_word)
+{
+	*config_word = *config_word | ((uint32_t)(config_number & 0x07) <<
+				       (channel * 3));
+}
+
+/**
+ * @brief Transmit 24-bit (3 bytes) of configuration information and reads back
+ * 24 bytes of data (3 bytes / 24-bit for each channel).
+ * @param dev - Device handler.
+ * @param config_word - 3 bytes of configuration data for 8 channels.
+ * @param data_array - Data array to read in 24 bytes of data from 8 channels.
+ * @return 0 in case of success, -1 otherwise.
+ */
+int32_t ltc2358_write_and_read(struct ltc2358_dev *dev, uint32_t config_word,
+			       uint8_t data_array[24])
+{
+	uint8_t tx_buff[24];
+	uint8_t i;
+
+	tx_buff[0] = (uint8_t)(config_word >> 16);
+	tx_buff[1] = (uint8_t)(config_word >> 8);
+	tx_buff[2] = (uint8_t)(config_word);
```

Comment:
Thanks for the review Ciprian. I have a concern regarding `no_os_put_unaligned_be24()` function, the buf[1] element array on the function would only store either 1 or 0 because of logical operation. Since I want to transmit 3 bytes of config information, I think that function would not be suitable for the operation.

Reply:
Fixed, thanks for reporting. Now the function I've mentioned should work as expected.